src/test/java/org/opentripplanner/gtfs/mapping/LocationMapperTest.java

```diff
@@ -22,18 +28,73 @@ static Stream<Arguments> testCases() {
   @ParameterizedTest(name = "a name of <{0}> should set bogusName={1}")
   @MethodSource("testCases")
   void testMapping(String name, boolean isBogusName) {
-    var gtfsLocation = new Location();
-    gtfsLocation.setId(new AgencyAndId("1", "zone-3"));
-    gtfsLocation.setName(name);
-    gtfsLocation.setGeometry(
-      new Polygon(
-        List.of(new LngLatAlt(1, 1), new LngLatAlt(1, 2), new LngLatAlt(1, 3), new LngLatAlt(1, 1))
-      )
-    );
+    var gtfsLocation = getLocation(name, Polygons.OSLO);
 
-    var mapper = new LocationMapper(StopModel.of());
+    var mapper = new LocationMapper(StopModel.of(), DataImportIssueStore.NOOP);
     var flexLocation = mapper.map(gtfsLocation);
 
     assertEquals(isBogusName, flexLocation.hasFallbackName());
   }
+
+  @Test
+  void selfIntersectingPolygon() {
+    var selfIntersecting = Polygons.SELF_INTERSECTING;
+    assertFalse(selfIntersecting.isValid());
+
+    var gtfsLocation = getLocation("invalid", selfIntersecting);
+
+    var issueStore = new DefaultDataImportIssueStore();
+    var mapper = new LocationMapper(StopModel.of(), issueStore);
+
+    mapper.map(gtfsLocation);
+
+    assertEquals(
+      List
+        .of(
+          Issue.issue(
+            "InvalidFlexAreaGeometry",
+            "GTFS flex location 1:zone-3 has an invalid geometry: Self-intersection at (lat: 1.0, lon: 2.0)"
+          )
+        )
+        .toString(),
+      issueStore.listIssues().toString()
+    );
+  }
+
+  @Test
+  void nonClosedPolygon() {
```

Comment:
Can these be called open polygons? I tried quickly reading about this and maybe the open is a bit vague so lets stick with nonClosed.

Reply:
We talked about this last week and the handling of the incomplete polygons was removed from this PR.